app/controllers/ht_contact_types_controller.rb

```diff
@@ -50,6 +54,10 @@ def destroy
 
   private
 
+  def log(params = @contact_type_params)
```

Comment:
I started to wonder if there was something clever we could do to define this once at the application controller and infer the correct object & params, but I think doing it the way you are rather than something clever and more Rails-like is  preferable.